### PR TITLE
drivers: sensors: lis2dh: Add ability to read temperature

### PR DESCRIPTION
**Pull request by ahedin25**
Allow user to read temperature from LIS2DH family of sensors.

Signed-off-by: Andrew Hedin <andrew.hedin@lairdconnect.com>

Tested with LIS2DH12 on BT510 board.  Tested with LIS3DH on MG100.

### CODE REVIEW

**Comment by avisconti** [comment]
> @MaureenHelm Sorry. I broke the other review. I'm still trying to determine if DIE_TEMP or AMBIENT is more appropriate.

@ahedin25 
Please note that this is an accel sensor, and the temperature is used just to compensate the offset, it is not an environmental sensor. So, export it as DIE_TEMP.

**Review by avisconti — CHANGES_REQUESTED** [review]
This driver is common for following sensors: LIS2DH, LIS3DH, LSM303DLHC,  LIS2DH12, LSM303AGR. In case of LIS2DH12 (for example) the raw temperature can be 8-bit ot 10-bit wide, depending on LPen. You cannot always assume it is 8-bit wide.

**Comment by ahedin25** [comment]
It seems that the following need to be added: a boolean to the DTS and the CTRL_REG1 (LPen) needs to be read first.

**Comment by ahedin25** [comment]
The fractional bits aren't important for my application and I am unsure about their usefulness in general.

It looks like the LSM303DLHC is 12 bits.

Should I add to the DTS to control the number of bits? Or do you envision something else?

**Review by avisconti — CHANGES_REQUESTED** [review]
It seems a great job, but a couple of request more from my side.

**Review by avisconti — CHANGES_REQUESTED** [review]
Also, add in the PR description on which sensor you tested it. Thanks!!

File: drivers/sensor/lis2dh/Kconfig [hunk 1]
```diff
@@ -138,4 +138,18 @@ config LIS2DH_ODR_9_LOW
...
+	bool "Enable temperature measurements"
+	select LIS2DH_BLOCK_DATA_UPDATE
+	help
+	  The temperature sensor is most suitable for measuring
+	  relative temperature. For example, it is X degrees C
+	  cooler or warmer.
+	  Each chip has an offset.  This offset must be applied
+	  to the result. The offset can be obtained by comparing the
+	  the reported temperature to a reference.
+
```
[the remark this answers is not in the document]
Comment (avisconti) [reply drivers/sensor/lis2dh/Kconfig:155]: I think it is good enough

**Review by avisconti — APPROVED** [review]

**Review by MaureenHelm — APPROVED** [review]
Please rebase to pick up a fix for the doc build error

**Pull request merged**